File: src/tools/cancelRequest.ts
import { mapUnknownError } from '../errors.js';
import { normalizeCancelRecord } from '../normalize.js';
import { HiggsfieldProviderClient } from '../provider.js';
import type { CancelRequestInput, CancelToolResponse } from '../types.js';

export async function cancelRequest(input: CancelRequestInput): Promise<CancelToolResponse> {
  try {
    const client = new HiggsfieldProviderClient();
    const record = await client.cancelRequest(input.request_id);
    return normalizeCancelRecord(record);
  } catch (error) {
    const mapped = mapUnknownError(error);
    return {
      request_id: input.request_id,
      cancelled: false,
      provider_status: mapped.code,
    };
  }
}

File: handlers/cancel_request.py
from __future__ import annotations

from clients.higgsfield_client import HiggsfieldProviderClient
from core.errors import map_unknown_error
from core.normalize import normalize_cancel_record
from models.inputs import CancelRequestInput
from models.outputs import CancelToolResponse


def handle_cancel_request(input: CancelRequestInput) -> CancelToolResponse:
    try:
        client = HiggsfieldProviderClient()
        record = client.cancel_request(input.request_id)
        return normalize_cancel_record(record)
    except Exception as error:
        mapped = map_unknown_error(error)
        return CancelToolResponse(
            request_id=input.request_id,
            cancelled=False,
            provider_status=mapped.code,
        )
